### ISSUE TITLE
openCV versions vastly differ in size

### ISSUE THREAD

**Issue by ha-noi** [opencv/opencv#6361]
This is a template helping you to create an issue which can be processes as quickly as possible. Feel free to add additional information or remove not relevant points if you do not need them.

If you have a question rather than reporting a bug please go to http://answers.opencv.org where you get much faster responses.

### Please state the information for your system
- OpenCV version: 2.4 / 3.x
- Host OS: Windows 8

### In which part of the OpenCV library you got the issue?
installation

Two different methods are discussed for installing openCV here
http://mathalope.co.uk/2015/05/07/opencv-python-how-to-install-opencv-python-package-to-anaconda-windows/

this is the 'standard' way of installing openCV
conda install -c https://conda.binstar.org/menpo opencv

but an alternative way is here
basically,  download the openCV binary and copy the cv2.pyd file to the Anaconda\Lib\site-packages directory.

The standard way puts a cv2.pyd file that is 2.0 to 2.4MB (depends on the version you pick)
The cv2.pyd from the download is 44MB!

This makes me a little uneasy. Is there any possibility of getting an explanation from this forum as to what accounts for the difference?

**Comment by alalek** [comment]
Try to open both files with "Dependency Walker" tool and you will see what library from conda requires a set of `opencv*.dll`, and the official library (with bigger size) is not (it includes all code in one file)

BTW, Don't miss ffmpeg wrapper DLL - it is required to read/write video files (there is no way to include code into the library directly due licensing issues).

**Comment by ha-noi** [comment]
I tried going to www.dependencywalker.com and got this error:

```
This site can’t be reached

www.dependencywalker.com took too long to respond.
ERR_CONNECTION_TIMED_OUT
```

**Comment by alalek** [comment]
You can try something from here: http://stackoverflow.com/questions/7378959/how-to-check-for-dll-dependency if you want some proofs that dependency lists are different.

**Comment by ha-noi** [comment]
thank you. I used dumpbin and got this

```
Dump of file F:\Anaconda\Lib\site-packages\cv2.pyd          
                                                            
File Type: DLL                                              
                                                            
  Image has the following dependencies:                     
                                                            
    python27.dll                                            
    MSVFW32.dll                                             
    AVIFIL32.dll                                            
    AVICAP32.dll                                            
    KERNEL32.dll                                            
    USER32.dll                                              
    GDI32.dll                                               
    ole32.dll                                               
    OLEAUT32.dll                                            
    COMDLG32.dll                                            
    ADVAPI32.dll                 
```                           
So now I am trying to figure out what you meant by 

> BTW, Don't miss ffmpeg wrapper DLL 

i installed ffmpeg into conda thusly:

> conda install --channel https://conda.anaconda.org/menpo ffmpeg

Interestingly there is no package for ffmpeg in anaconda/lib/site-packages
but there is a \Anaconda\pkgs\ffmpeg-2.7.0-0\Scripts folder with ffmpeg.exe in it

**Comment by alalek** [comment]
I mean OpenCV's opencv_ffmpeg.dll / opencv_ffmpeg64.dll files. More info is here: https://github.com/Itseez/opencv/tree/master/3rdparty/ffmpeg

**Comment by mshabunin** [comment]
If you take a look at the win-64/2.4.11 archive contents from here: https://anaconda.org/menpo/opencv/files , you will find cv2.pyd (2 Mb) + several opencv_*.dll (>21 Mb). Official release contains single cv2.pyd combined from all these files.
